cdk/lib/constructs/api.ts

```diff
@@ -104,8 +104,20 @@ export class BackendApi extends cdk.Construct {
     topicarea.addMethod("DELETE", apiIntegration, methodProps);
 
     const datasets = this.api.root.addResource("dataset");
+    datasets.addMethod("GET", apiIntegration, methodProps);
     datasets.addMethod("POST", apiIntegration, methodProps);
 
+    const dataset = datasets.addResource("{id}");
+    dataset.addMethod("GET", apiIntegration, methodProps);
+
+    const ingestApi = this.api.root.addResource("ingestapi");
+    const ingestApiDatasets = ingestApi.addResource("dataset");
+    ingestApiDatasets.addMethod("POST", apiIntegration, methodProps);
+
+    const ingestApiDataset = ingestApiDatasets.addResource("{id}");
+    ingestApiDataset.addMethod("PUT", apiIntegration, methodProps);
+    ingestApiDataset.addMethod("DELETE", apiIntegration, methodProps);
```

Comment:
This is where Authentication comes into place. This is fine for now, but we should change this to `API KEY` authentication in another Pull Request.